### PR TITLE
Code Example to store sensor data to mysql database

### PR DESCRIPTION
**Pull request by sigdelsanjog**
This code stores the continuous data received from the accelerometer to MySQL database with the help of mysql-python-connector at the interval of 3seconds(which can be changed as per the requirement)

### CODE REVIEW

**Comment by ladyada** [comment]
@sigdelsanjog thanks - please run `black` to format the code

File: examples/project/callibrate_data.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# """Code for maintaining the data before storing it to the database"""
+# class DataCallibration():
```
Comment (FoamyGuy) [inline examples/project/callibrate_data.py]: This looks to be commented out. If it's not needed lets remove it. If it is still needed please un-comment and show it's usage in one of the examples.

**Review by FoamyGuy — CHANGES_REQUESTED** [review]
This is passing the Black format check now. I think this is a great functional examples. It may be nice ultimately to make similar examples for other common sensors. 

However there are still some PyLint issues to work through. 

If I understand correctly this example is only going to work with Blinka / SB-PC devices right now because it's relying on `mysql.connector` which has no Circuit Python equivalent that I know of. 

If that is the case I think we should add a comment at the top that mentions it only works on Blinka / RasPi devices. 

@sigdelsanjog if you can make this changes that would be wonderful. If you need help any help feel free to ask and I can help out. If not that is okay as well, I will try to swing back around soon and work on getting this passing.

**Review by sigdelsanjog — COMMENTED** [review]
The code is good to go from my end. Admin can review it and merge to the master branch as an example to store data retrieved from the sensor to MySQL database.

@FoamyGuy

File: examples/project/callibrate_data.py (hunk 1 above)
Comment (sigdelsanjog) [reply examples/project/callibrate_data.py]: Made necessary changes.
Admin can delete callibrate_data.py

**Comment by sigdelsanjog** [comment]
@FoamyGuy 
Please review the code, and continue with the merge if the code seems okay to you.
Thank You

**Comment by FoamyGuy** [comment]
@sigdelsanjog We will need to get the formatting and pylint issues taken care of before we can merge it. they are shown on this page:  https://github.com/adafruit/Adafruit_CircuitPython_MPU6050/pull/9/checks

There is a learn guide here: https://learn.adafruit.com/improve-your-code-with-pylint that covers the process if you are willing to try tackling that. If you run into any trouble I can try to help you work through them.

I can come back and work through the formatting and pylint issues myself if you are unable to, but it will take me a bit of time to get back to it. Probably next week some time.

**Comment by sigdelsanjog** [comment]
@FoamyGuy I couldn't resolve the issue mentioned in [Adafruit_CircuitPython_MPU6050](https://github.com/adafruit/Adafruit_CircuitPython_MPU6050/pull/9/checks)
Please do the needful when you are back in your time convenience.

**Comment by evaherrada** [comment]
@sigdelsanjog Run `black --target-version=py35 .`

**Comment by FoamyGuy** [comment]
I tried out this code today. I was able to get mariadb (mysql) set up on the Raspberry Pi and run this example but it is throwing this error:
```
1146 (42S02): Table 'sensor_data.collector' doesn't exist
```
I think we need to add to this example the initialization of the db so that it is ready to receive data. @sigdelsanjog if you can add this it would be great. If not I can try to do that but I'm not very familiar with mysql and it may take me a while to have get to it.

**Comment by sigdelsanjog** [comment]
> I tried out this code today. I was able to get mariadb (mysql) set up on the Raspberry Pi and run this example but it is throwing this error:
> 
> ```
> 1146 (42S02): Table 'sensor_data.collector' doesn't exist
> ```
> 
> I think we need to add to this example the initialization of the db so that it is ready to receive data. @sigdelsanjog if you can add this it would be great. If not I can try to do that but I'm not very familiar with mysql and it may take me a while to have get to it.

@FoamyGuy Good to hear that you tried it on your PI.
Quick comments here:
1. Did you try executing the code after creating a database "sensor_data", creating a table collector within the database?
2. Did you also modify the host, username, and password to your local configuration?
My configuration points to a remote host in my home network (192.168.100.7)

Please try the following query to create the collector table:
--
-- Database: `sensor_data`
--

-- --------------------------------------------------------

--
-- Table structure for table `collector`
--

CREATE TABLE `collector` (
  `acc_x` float DEFAULT NULL,
  `acc_y` float DEFAULT NULL,
  `acc_z` float DEFAULT NULL,
  `created_at` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP
) ENGINE=InnoDB DEFAULT CHARSET=latin1;


 After you create a database and table collector, try executing the code. Please let me know if any error arises after that.

**Review by siddacious — CHANGES_REQUESTED** [review]
Hello @sigdelsanjog, thank you for your contribution. I'm sure others will find it useful. My apologies for not giving my input sooner.

Unfortunately this example will not run correctly unless a full mysql database is installed and configured according to the expectations of the example.

I think it would be better to instead update the examples to use `sqlite3` and to load a SQLite database file that you include with the PR. Anyone who is familiar enough with SQL to care about the difference will know how to set up a regular MySQL database and change the examples to use it, and those that don't will be able to use the example as-is with minimal changes.

Lastly, it looks like you renamed or moved files because `mpu6050_store_to_database.py` has a `DatabaseConnect` class, and `mysql_connect` has a `StoreToDatabase` class. Please fix this to be more clear.

I can imagine many cases where storing readings to a database will be useful so I look forward to seeing you update the code as I instructed above. Thanks again for your contribution.

**Comment by kattni** [comment]
@sigdelsanjog Thank you for the contribution. Have you had a chance to follow up with siddacious' review? Please let me know what your status is. Thanks!

**Comment by sigdelsanjog** [comment]
> @sigdelsanjog Thank you for the contribution. Have you had a chance to follow up with siddacious' review? Please let me know what your status is. Thanks!

@kattni 
I wrote this code for connecting with MySQL database. If one creates a MySQL database and updates the database settings the code works fine.
I was unable to update the code i.e. changing the database from MySQL to sqlite. This code was an outcome of my project so I thought of contributing to the repo example section.

There are some pylint issues which has caused the build fail.
If someone from the repo contributors could update/fix this, it would be great.
 For those who are looking to connect their Sensors and retrieve data via MySQL database, this code will be helpful to begin with.

In future someone might contribute the code to communicate with sqlite and other databases. Why not add them as well in the example section?

**Comment by jposada202020** [comment]
Hello. @kattni @sigdelsanjog 
At this moment what would be the future path with this PR?
1. If my understanding is correct, this could not be included as a standalone example as it is, as @FoamyGuy could not make it work
2. Siddacious request changes that has not been integrated in the PR
3. CI has not passed the test and work needs to be done to correct this.
Let me know your thoughts

Thanks

**Comment by FoamyGuy** [comment]
@kattni I think this could be closed for now. If the original contributor or anyone else wants to submit a more self-contained  example like Siddacious described in the future they could open a new PR for that.

**Comment by kattni** [comment]
Closing.

**Pull request closed without merging**